src/adapter/etl-adapter-doctrine/src/Flow/ETL/Adapter/Doctrine/DbalLimitOffsetExtractor.php

```diff
@@ -13,52 +14,51 @@
 
 final class DbalLimitOffsetExtractor implements Extractor
 {
-    /**
-     * @param array<OrderBy> $orderBy
-     */
     public function __construct(
         private readonly Connection $connection,
-        private readonly Table $table,
-        private readonly array $orderBy,
+        private readonly QueryBuilder $queryBuilder,
```

Comment:
I must say I'm not a huge fan of this solution, there is already an existing `DbalQueryExtractor` that should allow you to pass custom query, the whole purpose of using DbalLimitOffsetExtractor was to simplify the most common use case where you just need to extract a single table.

Reply:
But does `DbalQueryExtractor` support chunking?

Reply:
yes, its all about how you generate parameters

Reply:
Well, but you have to manually query for count and manually prepare all offset-limit pairs? Looks like a lot of userland code for adding one `join` to query :)

Reply:
So, WDYT? Because this is real requirement from real code using this library, if you don't want to merge such extractor into lib I can make external lib for that, but I really think that it is extremely useful to be able to write such queries in extractor with zero to none additional dependency (some dbal classes were already imported in this class anyway).

Reply:
I have an idea, it would be a BC break but might be worth it.
What we can do is to follow your approach with QueryBuilder but then add some simple static constructor that would prepare query builder in the way current constructor works. 
This way whoever will need to iterate over simple table will use static constructor and more advanced use cases would require custom query builder.